fastecho_module.c

```diff
@@ -43,6 +44,23 @@ static int fastecho_init_module(void)
 
     param.listen_sock = listen_sock;
 
+    /*
+     * Create a dedicated workqueue instead of using system_wq
+     * since the work item could be a CPU-intensive work item
+     * if its lifetime of connection is too long, e.g., using
+     * `telnet` to communicate with fastecho. Flag WQ_UNBOUND
+     * fits this scenario. Note that trade-off of this flag is
+     * cache locality. **If you are intend to benchmark** the
+     * module instead of using telnet-like program to interact
```

Comment:
Instead of specifying flags, can we detect the type of requests in advance?

Reply:
How about using module parameter to determine the workqueue flag? If the parameter doesn't being specified at module insertion, we use `WQ_UNBOUND` as the flag, and `0` vice versa.

Impl:
```c
static bool bench = false;
module_param(bench, bool, S_IRUGO);
...
fastecho_wq = alloc_workqueue(MODULE_NAME, bench ? 0 : WQ_UNBOUND, 0);

```

Reply:
By the way, [kselftest](https://www.kernel.org/doc/html/latest/dev-tools/kselftest.html) works pretty well on exercise individual code paths in the kernel. By adding parameter `bench` (or `batch`?) would automate internal tests as well.